### PR TITLE
WTCT: num_cores removed

### PR DESCRIPTION
**Pull request by tworec**
possible thanks to golemfactory/golem#3768

built on top of #334

### CODE REVIEW

File: golem_messages/message/tasks.py [hunk 1]
```diff
@@ -234,25 +234,52 @@ def __init__(self, *args, **kwargs):
 
 @library.register(TASK_MSG_BASE + 1)
 class WantToComputeTask(ConcentEnabled, base.Message):
+    """
+    A computation Offer.
+
+    A Provider sends it directly to a Requestor as a response to the Requestor's
+    Demand (Task) in order to get work (SubTask) to do.
```
Comment (shadeofblue) [inline golem_messages/message/tasks.py]: I'd add that the subtask is expressed as a `TaskToCompute`

File: golem_messages/message/tasks.py [hunk 2]
```diff
@@ -234,25 +234,52 @@ def __init__(self, *args, **kwargs):
...
+
+    Attributes:
+        node_name:          Provider's node name
+        perf_index:         Provider's Performance index
+        max_resource_size:  Provider's Storage size available for computation
+        max_memory_size:    Provider's RAM
+        price:              Offered price in GNT "WEI" (10e-18)
```
Comment (shadeofblue) [inline golem_messages/message/tasks.py]: I guess, we should define the unit here correctly ... I'm not completely sure now but I think that the price is expressed as "gnt wei / hour"

File: golem_messages/message/tasks.py [hunk 3]
```diff
@@ -234,25 +234,52 @@ def __init__(self, *args, **kwargs):
...
+        provider_ethereum_public_key:
+                            for transactions on ETH blockchain
+        task_header:        Demand; signed by a Requestor
+    """
     __slots__ = [
         'node_name',
         'perf_index',
         'max_resource_size',
         'max_memory_size',
-        'num_cores',
```
Comment (shadeofblue) [inline golem_messages/message/tasks.py]: is this indeed unused?

File: golem_messages/validators.py [hunk 4]
```diff
@@ -32,6 +32,16 @@ def validate_integer(field_name, value):
         )
 
 
+def validate_positive_integer(field_name, value):
+    validate_integer(field_name, value)
+    if value < 1:
+        raise exceptions.FieldError(
+            "Should be an positive integer",
```
Comment (shadeofblue) [inline golem_messages/validators.py]: s/an/a/

File: tests/message/test_tasks.py [hunk 5]
```diff
@@ -46,28 +46,74 @@ def test_extra_data(self):
...
+            exceptions.FieldError,
+            r"Should be an positive integer \[num_subtasks:-7\]"
+        ):
+            shortcuts.load(serialized, None, None)
+
+    def test_num_subtasks_int(self):
+        wtct = self.FACTORY()
+        self.assertIsInstance(wtct.num_subtasks, int)
+
+    def test_has_header_aka_check_proper_inheritance(self):
```
Comment (shadeofblue) [inline tests/message/test_tasks.py]: is there a reason we're adding this check to _this_ message in particular?

File: tests/message/test_tasks.py [hunk 6]
```diff
@@ -46,28 +46,74 @@ def test_extra_data(self):
...
+        from golem_messages.datastructures import MessageHeader
+        wtct = self.FACTORY()
+        self.assertIsNotNone(wtct.header)
+        self.assertIsInstance(wtct.header, MessageHeader)
+
```
Comment (shadeofblue) [inline tests/message/test_tasks.py]: I think I see no positive test ... iow, a test that tests a correctly set `num_subtasks` > 1 ...

**Comment by shadeofblue** [comment]
@tworec I think it's okay - just please ensure that it can be cleanly integrated with Golem Core and if changes in Golem Core are still needed, please ensure that a corresponding pull request to Golem Core is ready before merging this one.

According to new flow, we also need to give CodePoets heads up so they can also prepare a corresponding pull request for the Concent before we bump Golem Messages version in develop...

File: golem_messages/message/tasks.py (hunk 3 above)
Comment (tworec) [reply golem_messages/message/tasks.py]: yup, see golemfactory/golem#3768

File: tests/message/test_tasks.py (hunk 5 above)
Comment (tworec) [reply tests/message/test_tasks.py]: yes. I was not sure how multi inheritance in python works and this thing is supposed there will be call to both parent classes of WTCT

File: tests/message/test_tasks.py (hunk 6 above)
Comment (tworec) [reply tests/message/test_tasks.py]: ok

File: golem_messages/message/tasks.py (hunk 1 above)
Comment (tworec) [reply golem_messages/message/tasks.py]: ok

File: golem_messages/message/tasks.py (hunk 2 above)
Comment (tworec) [reply golem_messages/message/tasks.py]: well, I'd rather say that GM should be agnostic of such detail as payment scheme, isn't it?

File: golem_messages/validators.py (hunk 4 above)
Comment (tworec) [reply golem_messages/validators.py]: ok

File: tests/message/test_tasks.py (hunk 5 above)
Comment (shadeofblue) [reply tests/message/test_tasks.py]: ah :D

File: golem_messages/validators.py [hunk 7]
```diff
@@ -26,7 +26,7 @@ def validate_varchar(field_name, value, max_length):
 def validate_integer(field_name, value):
     if not isinstance(value, int):
         raise exceptions.FieldError(
-            "Should be an integer",
+            "Should be a integer",
```
Comment (shadeofblue) [inline golem_messages/validators.py]: haha, here, it should actually be an `an` ;)

File: golem_messages/message/tasks.py [hunk 8]
```diff
@@ -234,25 +234,52 @@ def __init__(self, *args, **kwargs):
...
         'price',
-        'concent_enabled',  # Provider notifies requestor
-                            # about his concent status
+        'num_subtasks',
+        'concent_enabled',
+        'extra_data',
+        # TODO: `provider` prefix is redundant; all above fields refers Provider
```
Comment (maaktweluit) [inline golem_messages/message/tasks.py]: Please add an issue number to a new TODO comment

Comment (tworec) [reply golem_messages/message/tasks.py]: there is no such issue, should I create one?

Comment (tworec) [reply golem_messages/message/tasks.py]: sorry, do not know how to go through those chained PR better. Now it's obvious I went somehow wrong

**Pull request merged**